documentation/modules/managing/con-consumer-config-properties.adoc

```diff
@@ -0,0 +1,252 @@
+// This module is included in the following files:
+//
+// assembly-client-config.adoc
+
+[id='con-consumer-config-properties-{context}']
+= Kafka consumer configuration tuning
+
+Use a basic consumer configuration with optional properties that are tailored to specific use cases.
+
+When tuning your consumers your primary concern will be ensuring that they cope efficiently with the amount of data ingested.
+As with the producer tuning, be prepared to make incremental changes until the consumers operate as expected.
+
+== Basic consumer configuration
+
+Connection and deserializer properties are required for every consumer.
+Generally, it is good practice to add a client id for tracking.
+
+In a consumer configuration, irrespective of any subsequent configuration:
+
+* The consumer fetches from a given offset and consumes the messages in order, unless the offset is changed to skip or re-read messages.
+* The broker does not know if the consumer processed the responses, even when committing offsets to Kafka, because the offsets might be sent to a different broker in the cluster.
+
+[source,shell,subs="+quotes,attributes"]
+----
+# ...
+bootstrap.servers=localhost:9092 <1>
+key.deserializer=org.apache.kafka.common.serialization.StringDeserializer  <2>
+value.deserializer=org.apache.kafka.common.serialization.StringDeserializer  <3>
+client.id=my-client <4>
+group.id=my-group-id <5>
+# ...
+----
+<1> (Required) Tells the consumer to connect to a Kafka cluster using a _host:port_ bootstrap server address for a Kafka broker.
+The consumer uses the address to discover and connect to all brokers in the cluster.
+Use a comma-separated list to specify two or three addresses in case a server is down, but it is not necessary to provide a list of all the brokers in the cluster.
+If you are using a loadbalancer service to expose the Kafka cluster, you only need the address for the service because the availability is handled by the loadbalancer.
+<2> (Required) Deserializer to transform the bytes fetched from the Kafka broker into message keys.
+<3> (Required) Deserializer to transform the bytes fetched from the Kafka broker into message values.
+<4> (Optional) The logical name for the client, which is used in logs and metrics to identify the source of a request. The id can also be used to throttle consumers based on processing time quotas.
+<5> (Conditional) A group id is _required_ if the consumer belongs to a consumer group.
```

Comment:
The sentence suggests that it is possible to have a consumer without a group, which is not possible if I'm not wrong and according to this 
https://github.com/apache/kafka/pull/362/commits/093e8e71c5fa158f5b8816e408a1d17974014f59

Reply:
AFAIK the `group.id` is not mandatory. You can have a consumer which is not part of any consumer group. You just cannot join a group without the `group.id`.

Reply:
I've changed the description to make this point clearer.

> <5> (Conditional) A group id is _required_ for a consumer to be able to join a consumer group